src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -26,4 +31,61 @@ protected Sections() {
 
         return stations;
     }
+
+    public void addAndResizeDistanceBy(Section section) {
+        resizeNearSection(section);
+
+        sections.add(section);
+    }
+
+    public boolean isAddable(Station upStation, Station downStation, Distance distance) {
+        if (sections.isEmpty()) {
+            return true;
+        }
+
+        return !containsBetweenDownStationAndDistance(downStation, distance)
+                && !containsBetweenUpStationAndDistance(upStation, distance)
+                && !containsStationsExactly(upStation, downStation)
+                && containsStationAny(upStation, downStation);
+    }
+
+    public SortedSections getSortedSections() {
+        return new SortedSections(this.sections);
+    }
+
+    private void resizeNearSection(Section section) {
+        sections.stream()
+                .filter(item -> item.isSameUpStation(section) || item.isSameDownStation(section))
+                .findFirst()
+                .ifPresent((near) -> near.resizeAndChangeNearStation(section));
+    }
+
+    private boolean containsBetweenDownStationAndDistance(Station downStation, Distance distance) {
+        return sections.stream()
+                .filter(item -> item.isDownStationBetween(downStation, distance))
+                .count() > 0L;
+    }
+
+    private boolean containsBetweenUpStationAndDistance(Station upStation, Distance distance) {
+        return sections.stream()
+                .filter(item -> item.isUpStationBetween(upStation, distance))
+                .count() > 0L;
```

Comment:
단순히 `exists` 여부만 확인하려면 아래와 같이도 표현 가능합니다 😄 
```suggestion
                return sections.stream()
                     .anyMatch(item -> item.isDownStationBetween(downStation, distance));
```

Reply:
이런방법이!! 고맙습니다 :)